src/samples.jl

```diff
@@ -394,24 +396,40 @@ function decode!(result_storage, sample_resolution_in_unit, sample_offset_in_uni
 end
 
 """
-    decode(samples::Samples, ::Type{T}=Float64)
-
-If `samples.encoded` is `true`, return a `Samples` instance that wraps
-
-    decode(convert(T, samples.info.sample_resolution_in_unit),
-           convert(T, samples.info.sample_offset_in_unit),
-           samples.data)
```

Comment:
I think this detail is still important here, since it affects the promotion that happens...